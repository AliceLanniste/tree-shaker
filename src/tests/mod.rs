fn tree_shake(input: &str) {
  let result = crate::tree_shake(input);
  let output = result.codegen_return.source_text;
  insta::assert_snapshot!(output);
}

#[test]
fn test_1() {
  tree_shake(
    r#"
      export let a = 1 && 2;
      export { b };
      let c = 3;
      let { ["b"]: b, d } = { b: 2, d: effect };
      "#,
  );
}
