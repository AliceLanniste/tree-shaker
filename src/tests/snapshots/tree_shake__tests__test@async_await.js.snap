---
source: src/tests/mod.rs
expression: tree_shake(&input)
input_file: src/tests/fixtures/async_await.js
---
export const l1 = 1;
export const l2 = "1a";
export const l3 = await unknown;
export const l4 = await unknown + "a";
async function pure() {
	return 1;
}
export const r1 = 1;
export const r2 = pure();
export const r3 = 1;
async function effect() {
	await something;
	return 1;
}
export const r4 = (await effect(), 1);
function effect2() {
	return something;
}
export const r5 = await effect2();
export async function pure_but_complex(unknown) {
	const f = async () => unknown;
	return await f();
}
